Automattic/wp-calypso
# Stepper: Onboarding flow update tracks events

Fixes: https://github.com/Automattic/wp-calypso/issues/94417
Fixes: https://github.com/Automattic/wp-calypso/issues/94420
Fixes: https://github.com/Automattic/wp-calypso/issues/94419
Fixes: https://github.com/Automattic/wp-calypso/issues/94418

## Proposed Changes

* Adds tracks event for plans and domain page step submit with correct props (https://github.com/Automattic/wp-calypso/issues/94417 and  https://github.com/Automattic/wp-calypso/issues/94420)
* Adds missing props for `calypso_signup_plan_select` when Plan is selected (`product_slug` and `from_section` )
* Adds `flow` prop to `calypso_page_view` event

## Why are these changes being made?


* Align with /start and be able to analyze important user data

## Testing Instructions



* Checkout branch or use live link
* Navigate to `/setup/onboarding`. Go through the flow and verify the correct events are recorded with necessary props.
* Compare to `/start` on production to see the latest state then compare against `/start` on local to ensure both are returning the same events and props.

## Pre-merge Checklist
